yt_dlp/extractor/adn.py

```diff
@@ -255,3 +290,42 @@ def _real_extract(self, url):
             'average_rating': float_or_none(video.get('rating') or metas.get('rating')),
             'comment_count': int_or_none(video.get('commentsCount')),
         }
+
+
+class ADNSeasonIE(ADNBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?(?:animation|anime)digitalnetwork\.(fr|de)/video/(?P<id>[^/]+)/?$'
+    _TESTS = [{
+        'url': 'https://animationdigitalnetwork.fr/video/tokyo-mew-mew-new',
+        'playlist_count': 12,
+        'info_dict': {
+            'id': 911,
+            'title': 'Tokyo Mew Mew New',
+        },
+        # 'skip': 'Only available in French end German speaking Europe',
+    }]
+
+    def _real_extract(self, url):
+        video_show_slug = self._match_id(url)
+        video_base_url = self._API_BASE_URL + 'show/%s/' % video_show_slug
+        show = self._download_json(
+            video_base_url, video_show_slug,
+            'Downloading show JSON metadata',
+            headers=self._HEADERS)['show']
+        show_id = show['id']
+        lang = self._get_distribution_language(url)
+        episodes = self._download_json(
+            self._API_BASE_URL + 'video/show/%s' % show_id, video_show_slug,
+            'Downloading episode list', headers={
+                'X-Target-Distribution': lang,
+            }, query={
+                'order': 'asc',
+                'limit': '-1',
+            })
+        entries = []
+        for episode in episodes['videos']:
+            entries.append(self.url_result(
+                'https://animationdigitalnetwork.%s/video/%s/%s' % (lang, video_show_slug, episode['id']),
+                ie=ADNIE.ie_key(),
+                video_id=episode['id']
+            ))
```

Comment:
```suggestion
        for episode_id in traverse_obj(episodes, ('videos', ..., 'id', {str})):
            entries.append(self.url_result(
                f'https://animationdigitalnetwork.{lang}/video/{video_show_slug}/{episode_id}',
                ADNIE, episode_id))
```
Add `from ..utils.traversal import traverse_obj` to the imports.

Reply:
didn't work with `{str}`

Reply:
Oh, those get returned as integers. Use `{str_or_none}` from `..utils`.

Reply:
Thanks, that works